api/src/main/java/com/github/eyefloaters/console/api/service/KafkaClusterService.java

```diff
@@ -105,33 +110,75 @@ public static Optional<Kafka> findCluster(SharedIndexInformer<Kafka> kafkaInform
                 .findFirst();
     }
 
-    public static Stream<ListenerStatus> externalListeners(Kafka kafka) {
-        return Optional.ofNullable(kafka.getStatus().getListeners())
-                .map(Collection::stream)
-                .orElseGet(Stream::empty)
-                .filter(l -> isExternalListener(kafka, l));
+    public static Optional<ListenerStatus> consoleListener(Kafka kafka) {
```

Comment:
Perhaps this should return an `Either` with the `Error` case explaining that the necessary criteria are?